src/libraries/Pool.sol

```diff
@@ -381,10 +381,17 @@ library Pool {
             // if the protocol fee is on, calculate how much is owed, decrement feeAmount, and increment protocolFee
             if (protocolFee > 0) {
                 unchecked {
-                    // step.amountIn does not include the swap fee, as it's already been taken from it,
-                    // so add it back to get the total amountIn and use that to calculate the amount of fees owed to the protocol
-                    // this line cannot overflow due to limits on the size of protocolFee and params.amountSpecified
-                    uint256 delta = (step.amountIn + step.feeAmount) * protocolFee / ProtocolFeeLibrary.PIPS_DENOMINATOR;
+                    uint256 delta;
+                    if (swapFee != protocolFee) {
+                        // step.amountIn does not include the swap fee, as it's already been taken from it,
+                        // so add it back to get the total amountIn and use that to calculate the amount of fees owed to the protocol
+                        // this line cannot overflow due to limits on the size of protocolFee and params.amountSpecified
+                        // this rounds down to favor LPs over the protocol
+                        delta = (step.amountIn + step.feeAmount) * protocolFee / ProtocolFeeLibrary.PIPS_DENOMINATOR;
+                    } else {
+                        // if the protocol fee is the same as the swap fee, the entire fee is owed to the protocol
+                        delta = step.feeAmount;
```

Comment:
also think i would rather this if statement written as
```
uint256 delta = (lpFee == 0)
   ? step.feeAmount
   : (step.amountIn + step.feeAmount) * protocolFee / ProtocolFeeLibrary.PIPS_DENOMINATOR
```

or lpFee != 0 and the other way around, whichever is cheaper probably

Reply:
oh yeah nice

Reply:
doesnt make a difference on gas either way